msteams-platform/toolkit/use-CICD-template.md

```diff
@@ -10,233 +10,359 @@ ms.date: 04/20/2022
+              --tenant ${{vars.AZURE_TENANT_ID}} \
+              --password ${{secrets.AZURE_SERVICE_PRINCIPAL_CLIENT_SECRET }} \
+              --interactive false
+    
+          - name: Deploy to hosting environment
+            run: |
+              npx teamsapp deploy --ignore-env-file true \
+              --interactive false
+    
+          - name: Package app
+            run: |
+              npx teamsapp package
+    
+          - name: upload appPackage
+            uses: actions/upload-artifact@v4
+            with:
+              name: artifact
+              path: appPackage/build/appPackage.zip
+    ```
+
+    > [!NOTE]
+    > The default pipeline triggers when push events occur on the main branch. You've the option to modify it to suit your specific requirements.
+
+1. Go to GitHub.
+
+1. Locate the following variables and secrets:
+
+    * `AZURE_SERVICE_PRINCIPAL_CLIENT_ID`, `AZURE_TENANT_ID`, and `AZURE_SERVICE_PRINCIPAL_CLIENT_SECRET`. [Need info.]
+
+      :::image type="content" source="../assets/images/teams-toolkit-v2/repo-settings.png" alt-text="Screenshot shows the repo settings.":::
+
+      > [!NOTE]
+      > The `AZURE_SERVICE_PRINCIPAL_CLIENT_SECRET` variable must be set as secret.
+      > Utilize the [GitHub environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#environment-variables) to use different sets of variables.
+
+    * `BOT_AZURE_APP_SERVICE_RESOURCE_ID`: Go to the `teamsapp.yml` file. In the `deploy` stage, the values enclosed in `${{}}` are the required variable keys. If you've used the `provision` command from Teams Toolkit, you can locate the values in the environment files in the `.env` folder.
+
+      :::image type="content" source="../assets/images/teams-toolkit-v2/teamsappyml.png" alt-text="Screenshot shows the bot Azure app service resource ID in teamsapp.yml file.":::
 
-### Customize CI workflow
+    * `TEAMS_APP_ID`: Go to the `appPackage/manifest.json` file. Go to `id`, the values enclosed in `${{}}` are the required variable keys. If you've used the `provision` command from Teams Toolkit, you can locate the values in the environment files in the `.env` folder.
 
-To customize the CI workflow, you can do the following:
+      :::image type="content" source="../assets/images/teams-toolkit-v2/manifest.png" alt-text="Screenshot shows the Teams app ID in manifest file.":::
 
-1. Change the trigger: By default, the CI workflow is triggered when a new pull request is created against `dev` branch.
-1. Add scripts to build the project: By default, the `Build the project` step is commented.
-1. Add scripts to run unit test: By default, the `Run unit test` step is commented.
+1. In the GitHub, navigate to your repository’s **Settings** and select **Secrets and variables** > **Actions**.
 
-### Customize CD workflow
+    Update the variable keys that you've gathered for the following variables:
```

Comment:
```suggestion
   For the above example, you need to set the following variables.
```

Reply:
Hi @v-bvishnu , please help update this too, thanks!

Reply:
We're informing the user to perform all the steps at the starting of the section:
to set up the pipeline with GitHub, follow these steps
. Hence, need not to mention the user saying that "
For the above example, you need to set the following variables. For your own project, please follow above steps to collect the needed variables."